test(e2e): pretty-print ExitError on cmd failure

antoineco opened the pull request:
<!--

Thank you for opening a new PR to the Unikraft Open Source Project!  We welcome
new changes, features, fixes, and more!  Please fill in this form to indicate
the status of your PR.  Please ensure you have read the contribution guidelines
before opening a new PR as this will cover the PR process:

  https://unikraft.org/docs/contributing/

  Kraftkit follows the same guidelines as the Unikraft Open Source Project.

-->

### Prerequisite checklist

<!--
Please mark items appropriately:
-->

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [x] Updated relevant documentation.

### Description of changes

Wraps `*exec.ExitError` in a type that implements `GomegaStringer`, so that these errors can be pretty-printed (in this particular case: dump stderr).

GitHub-Fixes: #415

Example:

```
kraft version when invoked with the --help flag [It] should print the command's help
/home/acotten/git/unikraft/kraftkit/test/e2e/cli/version_test.go:54

  [FAILED] Unexpected error:
      <*cmd.ExitError | 0xc00034a0c0>:
      exit status 2
      panic: invalid argument "<bool Value>" for "--fast" flag: strconv.ParseBool: parsing "<bool Value>": invalid syntax

          goroutine 1 [running]:
          kraftkit.sh/cmd/kraft/build.New()
                /home/acotten/git/unikraft/kraftkit/cmd/kraft/build/build.go:71 +0x20d
          main.New()
                /home/acotten/git/unikraft/kraftkit/cmd/kraft/kraft.go:69 +0x28a
          main.main()
                /home/acotten/git/unikraft/kraftkit/cmd/kraft/kraft.go:101 +0x2b

  occurred
```

antoineco committed:
test(e2e): pretty-print ExitError on cmd failure

GitHub-Fixes: #415

Signed-off-by: Antoine Cotten <antoine@unikraft.io>

nderjung reviewed (approved):
Tried it out locally. Works nicely. Thanks!

Reviewed-by: Alexander Jung <alex@unikraft.io>
Approved-by: Alexander Jung <alex@unikraft.io>

nderjung reviewed (changes requested):
Holding. Looks like there is an issue with how it's used in actions pipeline.

antoineco commented:
@nderjung I removed the `BeAssignableToTypeOf(exec.ExitError)` check + exit code assertion—which was super clunky anyway—and replaced it with `MatchError()`.

This works equally fine because Go logs `ExitError` using [a standard string](https://cs.opensource.google/go/go/+/refs/tags/go1.20.3:src/os/exec_posix.go;l=105-111), and is not reliant on the underlying Go type.

nderjung reviewed (approved):
All good, thanks!

Approved-by: Alexander Jung <alex@unikraft.io>
